crates/ide/src/hover.rs

```diff
@@ -117,7 +117,37 @@ pub(crate) fn hover(
     let mut seen = HashSet::default();
 
     let mut fallback = None;
-    sema.descend_into_macros_many(token.clone())
+    // attributes, require special machinery as they are mere ident tokens
+
+    let descend_macros = sema.descend_into_macros_many(token.clone());
+
+    for token in &descend_macros {
+        if token.kind() != COMMENT {
+            if let Some(attr) = token.ancestors().find_map(ast::Attr::cast) {
+                // lints
+                if let Some(res) = try_hover_for_lint(&attr, &token) {
+                    return Some(res);
+                // derives
+                } else {
+                    let def =
+                        try_resolve_derive_input_at(&sema, &attr, &token).map(Definition::Macro);
+                    if let Some(def) = def {
+                        if let Some(hover) = hover_for_definition(
+                            &sema,
+                            file_id,
+                            def,
+                            &token.parent().unwrap(),
+                            config,
+                        ) {
+                            return Some(RangeInfo::new(token.text_range(), hover));
+                        }
+                    }
+                }
+            }
+        }
+    }
```

Comment:
It's a bit unclear to me as to why this is being moved out of the main iteration

Reply:
Now derive token is in the find_definition (now Definition::from_node), and for lints it isn't a problem?

Reply:
I still don't really understand why you are pulling that one out of all the other logic. But it's fine I'm gonna refactor this module in the coming week so I can address that.